subctl/issue-183-support-upgrading-submariner.md

```diff
@@ -0,0 +1,63 @@
+# Adding command to subctl to upgrade Submariner
+
+Related Issue: [Add subctl upgrade command](https://github.com/submariner-io/enhancements/issues/183)
+
+## Summary
+
+Currently, there is no automated way to upgrade Submariner. The user needs to manually upgrade `subctl`, since the version of Submariner
+to be installed is tied with the version of `subctl` installed, and then re-run `deploy-broker` and `join` sub-commands. This proposal
+describes a way to automate this process.
+
+## Proposal
+
+The proposal is to automate the process by adding a new `upgrade` sub-command that will do the above steps in the sequence.
+
+## Design details
+
+* subctl command name: `upgrade`
+* Flags for the command:
+  * `to`: mentions the version Submariner should be upgraded to. If not specified, upgrade to the latest stable version available.
+* Context to run on:
+  * Check on all contexts and re-run `deploy-broker` and `join` on clusters where Broker and Submariner are installed respectively.
```

Comment:
Right now, the idea is to have all clusters in the setup at the same version. There might be some functionality issues if all clusters are not at the same version, say for example: gather collects more info from 0.16 cluster than from 0.15 cluster. When we look at the gather data while troubleshooting, it might cause issues or the info might not be enough to troubleshoot properly.